adserver/models.py

```diff
@@ -1227,6 +1228,14 @@ class AdImpression(BaseImpression):
     )
 
     class Meta:
+        # We must also constrain when the `advertisement` is null
+        constraints = (
+            UniqueConstraint(
```

Comment:
Cool, I didn't know Django could do this 👍

Reply:
It's new in 2.2.